connection.go

```diff
@@ -4,45 +4,51 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"errors"
+	"io"
 	"net"
 	"net/url"
 	"os"
 	"strings"
 	"time"
 
+	"github.com/cenkalti/backoff"
+	"github.com/go-kit/kit/log"
 	"golang.org/x/net/websocket"
 )
 
+const TheBigestMaxGremlinConnectionsLimit = 5
+
 // Clients include the necessary info to connect to the server and the underlying socket
 type Pool struct {
 	urlStr         string
 	origin         string
 	MaxConnections int
 	Auth           []OptAuth
-	// Connections    map[int]*Connection
-	Connections chan Connection
+	Connections    chan Connection
+	logger         log.Logger
 }
 
-const TheBigestMaxGremlinConnectionsLimit = 5
-
 type Connection struct {
 	id   int
 	ws   *websocket.Conn
 	busy bool
 	pool *Pool
 }
 
-func NewClient(urlStr string, origin string, maxConn []int, options ...OptAuth) (*Pool, error) {
-	// Check if connection is possible
-	_, err := websocket.Dial(urlStr, "", origin)
-	if err != nil {
-		return nil, err
-	}
+func NewClient(logger log.Logger, urlStr string, origin string, maxConn []int, options ...OptAuth) (*Pool, error) {
 	pool := &Pool{
 		urlStr: urlStr,
 		origin: origin,
 		Auth:   options,
+		logger: logger,
 	}
+
+	// Check if connection is possible and close it
```

Comment:
Why do we need to do this?

Reply:
It is for initial check if gremlin is reachable on gremlin client creation
but the same is done in a loop few rows down - so I removed it